xarray/backends/pnc_.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+
+import numpy as np
+
+from .. import Variable
+from ..core.utils import (FrozenOrderedDict, Frozen)
+from ..core import indexing
+
+from .common import AbstractDataStore, DataStorePickleMixin, BackendArray
+
+
+class PncArrayWrapper(BackendArray):
+
+    def __init__(self, variable_name, datastore):
+        self.datastore = datastore
+        self.variable_name = variable_name
+        array = self.get_array()
+        self.shape = array.shape
+        self.dtype = np.dtype(array.dtype)
+
+    def get_array(self):
+        self.datastore.assert_open()
+        return self.datastore.ds.variables[self.variable_name]
+
+    def __getitem__(self, key):
+        key = indexing.unwrap_explicit_indexer(
+            key, target=self, allow=indexing.BasicIndexer)
+
+        with self.datastore.ensure_open(autoclose=True):
+            array = self.get_array()
+            if key == () and self.ndim == 0:
+                return array[...]
+            return array[key]
+
+
+class PncDataStore(AbstractDataStore, DataStorePickleMixin):
+    """Store for accessing datasets via PseudoNetCDF
+    """
+
+    def __init__(self, filename, mode='r', autoclose=False):
+        from PseudoNetCDF import pncopen
+        try:
+            opener = functools.partial(pncopen, filename,
+                                       mode=mode, format='netcdf')
+            self.ds = opener()
+        except Exception:
```

Comment:
Why do you need all these fallbacks? This logic is usually best avoided -- it's better to require users to be explicit.

You should pick some default behavior for `open` (which will be used by `open_dataset`) and require explicitly opening the backend object if that does not suffice for some use-case.

Reply:
I can remove the first except wrap, but the second is present because not all constructors except the mode keyword. Individual constructors are wrapped by pncopen.

Reply:
Can you simply leave out the `mode` argument in that case? There's no particular reason xarray needs to include it, as long as the default behavior supports reads.

Reply:
I can leave out mode, but that will prevent the formats that do support mode from using it. I could add a `**kwds` option to `open` much like pncopen. The default behavior would, as you suggest, leave out mode.